test/e2e/scenario_basic_flow_test.go

```diff
@@ -0,0 +1,173 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+	"knative.dev/func/test/oncluster"
+	"knative.dev/func/test/testhttp"
+
+	common "knative.dev/func/test/common"
+)
+
+// TestFunctionBasicFlow will run a comprehensive path of func commands an end user may perform such as
+// > create > build > run > deploy > info > list > (curl) > invoke > deploy (new revision) > (curl) > delete > list
+func TestFunctionBasicFlow(t *testing.T) {
+
+	var funcName = "basic-function"
+	var funcPath = filepath.Join(t.TempDir(), funcName)
+
+	knFunc := common.NewKnFuncShellCli(t)
+	knFunc.ShouldDumpOnSuccess = false
+
+	// ---------------------------
+	// Func Create Test
+	// ---------------------------
+	knFunc.Exec("create", "--language", "node", "--template", "http", funcPath)
+
+	// From here on, all commands will be executed from the func project path
+	knFunc.SourceDir = funcPath
+
+	// ---------------------------
+	// Func Build Test
+	// ---------------------------
+	knFunc.Exec("build", "--registry", common.GetRegistry())
```

Comment:
A user should generally not need to/want to run 'build'.  This command is only used for the edge-case when a developer just wants the container to be built for some reason.  The `deploy` command should (almost) always do what they want:

Primary Flow:
`create`-> `deploy` -> `curl` -> `delete`
I would perhaps put the explicit 'build' command down near the bottom since it is an optional helper like `run`, `invoke` or `status`

Reply:
Hello Luke! Correct! Perhaps I can change the name of the scenario to "extended" or "complete" flow (the name `basic` trully does not reflect the purpose of this test).

The idea is to have one more comprehensive flow that walks thru developer local experience (build,run) as well as deployment on cluster experience (deploy, update, re-deploy, check,...).

It would also cover/replace previous Invoke test (by testing against a function running locally and deployed on cluster) as well as give at least one shot test other (less used) such as `list` and `info` (previously on runtime test).

The basic primary commands (`create` > `deploy` > `curl` > `delete`) as you mentioned I've been thinking in leave in the runtime (lifecycle) test, for both http and cloudevents. https://github.com/knative/func/pull/1615/commits/974a37132778d03e25c1b74899d7f9eb23ee3b96#diff-00b3b4a4cedefae054b320bcd68e937d94eb09caafa1dcde3877cf12fc484542R69 

I'm putting together a doc with the proposal and explaining these changes in comparison with existing tests and I shall share here soon.